src/transforms/closure.jl

```diff
@@ -13,14 +13,49 @@ struct Closure <: Transform end
 
 isrevertible(::Type{Closure}) = true
 
-function apply(::Closure, table)
-  # TODO
+assertions(::Type{Closure}) = [TT.assert_continuous]
+
+function apply(transform::Closure, table)
+  # basic checks
+  for assertion in assertions(transform)
+    assertion(table)
+  end
+
+  # original column names
+  names = Tables.columnnames(table)
+
+  # table as matrix and get the sum acros dims 2
+  X = Tables.matrix(table)
+  S = sum(X, dims=2)
+
+  # divides each row by its sum (closure operation)
+  Z = X ./ S
+
+  # table with the old columns and the new values
+  𝒯 = (; zip(names, eachcol(Z))...)
+  newtable = 𝒯 |> Tables.materializer(table)
+  
+  newtable, S
 end
 
 function revert(::Closure, newtable, cache)
-  # TODO
+  # transformed column names
+  names = Tables.columnnames(newtable)
+
+  # table as matrix
+  Z = Tables.matrix(newtable)
+
+  # retrieve cache
+  S = cache
+
+  # undo operation
+  X = Z .* S
+
+  # table with original columns
+  𝒯 = (; zip(names, eachcol(X))...)
+  𝒯 |> Tables.materializer(newtable)
 end
 
-function reapply(::Closure, table, cache)
-  # TODO
+function reapply(transform::Closure, table, cache)
+  apply(transform, table) # how to reuse cache (column of sum) to a (possibly different) table ?
```

Comment:
Good question. I think we should make Closure a `Stateless` transform and erase this `reapply` method. 👍🏽